Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -167,6 +166,11 @@
 #include <LibWeb/WebIDL/DOMException.h>
 #include <LibWeb/WebIDL/ExceptionOr.h>
 
+#ifdef AK_OS_WINDOWS
+#    include <AK/Windows.h>
+#    define localtime_r(time, tm) localtime_s(tm, time)
+#endif
```

Comment:
Oof, we should try to prevent these things in LibWeb. Instead of going through timespec and localtime_r, can't we add helper methods in AK instead?